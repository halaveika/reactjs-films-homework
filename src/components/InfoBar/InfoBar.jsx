import React from 'react';
import PropTypes from 'prop-types';
import './InfoBar.scss';

export default function InfoBar({
  original_title, genres, runtime, vote_average,
}) {
  return (
    <div className="info-container">
      <span className="title">
        {original_title}
      </span>
      <span className="genre">
        {genres.map((e) => e.name).join('  ')}
        {' '}
        |
        {runtime}
      </span>
      <div className="popularity">
        {Array(Math.round(vote_average)).fill(1).map((e, i) => <div className="star" key={i} />)}
        <span>
          {vote_average}
        </span>
      </div>
    </div>
  );
}

InfoBar.propTypes = {
  original_title: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  })).isRequired,
  runtime: PropTypes.number.isRequired,
  vote_average: PropTypes.number.isRequired,
};
