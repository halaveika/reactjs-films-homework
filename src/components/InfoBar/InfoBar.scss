.info-container {
  width: 300px;
  height: auto;
}

.title {
  font-size: 30px;
  font-weight: bold;
}

.genre {
  font-size: 16px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.popularity {
  display: flex;
  flex-direction: row;
  align-items: center;

  .star {
    width: 20px;
    height: 20px;
  }

  span {
    margin-left: 20px;
    font-size: 16px;
    border: #1890ff solid 2px;
    border-radius: 25%;
    padding: 1px 4px;
  }
}
