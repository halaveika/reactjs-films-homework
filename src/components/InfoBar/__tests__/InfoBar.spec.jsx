import React from 'react';
import renderer from 'react-test-renderer';
import InfoBar from '..';

const mockData = {
  original_title: 'the jungle book',
  genres: [{
    id: 13,
    name: 'Adventure',
  }, {
    id: 18,
    name: 'Drama',
  }, {
    id: 20,
    name: 'Family',
  }, {
    id: 21,
    name: 'Fantasy',
  }],
  vote_average: 4.8,
  runtime: 139,
  overview: 'There are growing dangers in the wizarding world of 1926 New York. Something mysterious is leaving a path of destruction in the streets, threatening to expose the wizarding',
};

describe('test InfoBar component', () => {
  const component = renderer.create(
    <InfoBar
      original_title={mockData.original_title}
      genres={mockData.genres}
      vote_average={mockData.vote_average}
      runtime={mockData.runtime}
    />,
  );
  const tree = component.toJSON();
  it('should render InfoBar component', () => {
    expect(tree).toMatchSnapshot();
  });
});
