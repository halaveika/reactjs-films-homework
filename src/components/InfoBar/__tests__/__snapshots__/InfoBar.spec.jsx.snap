// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test InfoBar component should render InfoBar component 1`] = `
<div
  className="info-container"
>
  <span
    className="title"
  >
    the jungle book
  </span>
  <span
    className="genre"
  >
    Adventure  Drama  Family  Fantasy
     
    |
    139
  </span>
  <div
    className="popularity"
  >
    <div
      className="star"
    />
    <div
      className="star"
    />
    <div
      className="star"
    />
    <div
      className="star"
    />
    <div
      className="star"
    />
    <span>
      4.8
    </span>
  </div>
</div>
`;
